sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3304,6 +3304,22 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession {
         """.stripMargin).collect()
     }
   }
+
+  test("SPARK-29860: Fix dataType mismatch issue for InSubquery") {
+    withTable("ta", "tb", "tc") {
+      sql("create table ta(id Decimal(18,0)) using parquet")
```

Comment:
To make this unit test faster, can we use a temporary table here?

Reply:
thanks, done.